src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -181,105 +181,144 @@ public BuildStepMonitor getRequiredMonitorService() {
+        setBuildName(buildName);
+        setIncludeBuildNumberInKey(includeBuildNumberInKey);
+        setProjectKey(projectKey);
+        setPrependParentProjectKey(prependParentProjectKey);
+        setDisableInprogressNotification(disableInprogressNotification);
+        setConsiderUnstableAsSuccess(considerUnstableAsSuccess);
+    }
 
-        this.stashServerBaseUrl = stashServerBaseUrl != null && stashServerBaseUrl.endsWith("/")
-                ? stashServerBaseUrl.substring(0, stashServerBaseUrl.length() - 1)
-                : stashServerBaseUrl;
-        this.credentialsId = credentialsId;
-        this.ignoreUnverifiedSSLPeer = ignoreUnverifiedSSLPeer;
-        this.commitSha1 = commitSha1;
-
-        StashBuildState overwrittenBuildState = null;
-        try {
-            overwrittenBuildState = StashBuildState.valueOf(buildStatus);
-        } catch (Exception e) {
-            // ignore unknown or null values
-        }
-        this.buildStatus = overwrittenBuildState;
-
-        this.buildName = buildName;
-        this.includeBuildNumberInKey = includeBuildNumberInKey;
-        this.projectKey = projectKey;
-        this.prependParentProjectKey = prependParentProjectKey;
-        this.disableInprogressNotification = disableInprogressNotification;
-        this.considerUnstableAsSuccess = considerUnstableAsSuccess;
+    StashNotifier(
+            JenkinsLocationConfiguration globalConfig
+    ) {
         this.globalConfig = globalConfig;
     }
 
     @DataBoundConstructor
-    public StashNotifier(
-            String stashServerBaseUrl,
-            String credentialsId,
-            boolean ignoreUnverifiedSSLPeer,
-            String commitSha1,
-            String buildStatus,
-            String buildName,
-            boolean includeBuildNumberInKey,
-            String projectKey,
-            boolean prependParentProjectKey,
-            boolean disableInprogressNotification,
-            boolean considerUnstableAsSuccess
-    ) {
+    public StashNotifier() {
         this(
-                stashServerBaseUrl,
-                credentialsId,
-                ignoreUnverifiedSSLPeer,
-                commitSha1,
-                buildStatus,
-                buildName,
-                includeBuildNumberInKey,
-                projectKey,
-                prependParentProjectKey,
-                disableInprogressNotification,
-                considerUnstableAsSuccess,
                 JenkinsLocationConfiguration.get()
         );
     }
 
-    public boolean isDisableInprogressNotification() {
-        return disableInprogressNotification;
+    public String getStashServerBaseUrl() {
+        return stashServerBaseUrl;
     }
 
-    public boolean isConsiderUnstableAsSuccess() {
-        return considerUnstableAsSuccess;
+    @DataBoundSetter
+    public void setStashServerBaseUrl(String stashServerBaseUrl) {
+        this.stashServerBaseUrl = stashServerBaseUrl != null && stashServerBaseUrl.endsWith("/")
```

Comment:
[StringUtils#stripEnd](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#stripEnd(java.lang.String,%20java.lang.String)) could be used.